content/en/blog/_posts/2025-05-09-start-sidecar-first/index.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: blog
+title: "Start Sidecar First: How To Avoid Snags"
+date: 2025-05-05
+draft: true
+slug: start-sidecar-first
+author: Agata Skorupka (The Scale Factory)
+---
+
+From the [previous blog post](/blog/2025/04/22/multi-container-pods-overview/) on Multicontainer Pods you know what their job is, what are the main architectural patterns, and how they are implemented in Kubernetes. The main thing I’ll cover in this article is how to ensure that your sidecar containers start before the main app. It’s more complicated than you might think!
+
+## A gentle refresher
+
+Let’s just remind that the [v1.29.0 release of Kubernetes](/blog/2023/12/13/kubernetes-v1-29-release/) added native support for [sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/), which can now be defined within the `spec.initContainers` field, but with `restartPolicy: Always` as in the example below (note that it is a partial snippet of the full Kubernetes manifest):
+
+```yaml
+initContainers:
+  - name: logshipper
+    image: alpine:latest
+    restartPolicy: Always
+    command: ['sh', '-c', 'tail -F /opt/logs.txt']
+    volumeMounts:
+    - name: data
+        mountPath: /opt
+```
+
+What’s the specifics of defining sidecars with an `spec.initContainers` block, rather than as a regular multi-container pod? `spec.initContainers` are always started first (before the main application) and terminated last (after the main application). Furthermore, when used with [Jobs](/docs/concepts/workloads/controllers/job/), sidecar would still be alive and potentially restarted after Job is done, and they will not block pod completion (see also [Pod sidecar containers tutorial](/docs/tutorials/configuration/pod-sidecar-containers/)).
```

Comment:
s/What’s the/What are the/

s/an `spec.initContainers`/a `spec.initContainers`/

> are always started first (before the main application) and terminated last (after the main application).

Consider simplifying and being more specific, For example:

`are always started before the main application and terminated after the main application.`

s/sidecar would still be alive and potentially restarted after Job.../a sidecar container is still alive and potentially restarts after the Job.../